### PR TITLE
Fix Docker Pipeline

### PR DESCRIPTION
**Pull request by justusschock**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## What does this PR do?
Fixes # (issue).

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: .github/workflows/docker_builds.yml [hunk 1]
```diff
@@ -15,22 +15,22 @@ jobs:
...
+    - name: Extract branch name
+      shell: bash
+      run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
+      id: extract_branch
     - uses: actions/checkout@v2
     - name: Publish Releases to Docker
       # only on releases
       uses: elgohr/Publish-Docker-Github-Action@2.14
-      if: contains(github.ref, 'refs/tags/') && !contains(${{ steps.get_version.outputs.VERSION }}, 'rc') %% !contains(${{ steps.get_version.outputs.VERSION }}, 'dev')
+      if: contains(github.ref, 'refs/tags/') && !contains(${{ steps.extract_branch.outputs.branch }}, 'rc') && !contains(${{ steps.extract_branch.outputs.branch }}, 'dev')
```
Comment (Borda) [inline .github/workflows/docker_builds.yml:26]: branch is not RC, shall it be rather tag?

Comment (justusschock) [reply .github/workflows/docker_builds.yml]: Fixed by parsing both (we need branches for nightlies) and I thought tags are on the same ref, but they aren't. Good catch!

File: .github/workflows/docker_builds.yml [hunk 2]
```diff
@@ -15,22 +15,27 @@ jobs:
...
+    - name: Extract Tag name
+      if: contains(github.ref, 'refs/tags')
+      shell: bash
+      run: echo "##[set-output name=tag;]$(echo ${GITHUB_REF#refs/tags/})"
     - uses: actions/checkout@v2
     - name: Publish Releases to Docker
       # only on releases
       uses: elgohr/Publish-Docker-Github-Action@2.14
-      if: contains(github.ref, 'refs/tags/') && !contains(${{ steps.get_version.outputs.VERSION }}, 'rc') %% !contains(${{ steps.get_version.outputs.VERSION }}, 'dev')
+      if: contains(github.ref, 'refs/tags/') && contains(${{ steps.extract_tag.outputs.tag }}, 'rc') != true && contains(${{ steps.extract_tag.outputs.tag }}, 'dev') != true
```
Comment (Borda) [inline .github/workflows/docker_builds.yml:31]:
```suggestion
      if: contains(github.ref, 'refs/tags/') && !contains(${{ steps.extract_tag.outputs.tag }}, 'rc') && !contains(${{ steps.extract_tag.outputs.tag }}, 'dev')
```
not sure, need to check it...

File: .github/workflows/docker_builds.yml [hunk 3]
```diff
@@ -15,22 +15,27 @@ jobs:
...
     steps:
-    - name: Extract Current Tag
-      if: contains(github.ref, 'refs/tags/')
-      id: get_version
-      run: echo ::set-output name=VERSION::${GITHUB_REF/refs\/tags\//}
+    - name: Extract branch name
+      if: contains(github.ref, 'refs/tags/') != true
+      shell: bash
+      run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
+      id: extract_tag
```
Comment (Borda) [inline .github/workflows/docker_builds.yml:22]: why this is needed?

File: .github/workflows/docker_builds.yml [hunk 4]
```diff
@@ -40,5 +45,5 @@ jobs:
         username: ${{ secrets.DOCKER_USERNAME }}
         password: ${{ secrets.DOCKER_PASSWORD }}
         dockerfile: docker/Dockerfile
-        buildargs: PYTHON_VERSION=${{ matrix.python_version }},PYTORCH_VERSION=${{ matrix.pytorch_version }},LIGHTNING_VERSION=${{ steps.get_version.outputs.VERSION }}
+        buildargs: PYTHON_VERSION=${{ matrix.python_version }},PYTORCH_VERSION=${{ matrix.pytorch_version }},LIGHTNING_VERSION=${{ steps.extract_branch.outputs.branch }}
```
Comment (Borda) [inline .github/workflows/docker_builds.yml:48]: still branch? I see that this run on `master` only

**Pull request merged**